### DISCUSSION TITLE
Deploying on AWS - Umbrella thread

### DISCUSSION THREAD

**Discussion by ennioVisco** [nitrojs/nitro#1106] [General]
Being AWS a fairly general cloud service, there are several ways to deploy on it:

### Packaging Nitro in Lambda-compatible functions & expose via API Gateway
It involves the usage of the current `aws-lambda` preset to build the functions, as described in https://nitro.unjs.io/deploy/providers/aws, and is a bit far from zero-config, as it still requires setting up:
-  an API Gateway
- the S3 bucket that will contain the functions

### Packaging the Nitro-based app for static hosting:
Of course, another option when possible for the application that is using Nitro, (e.g. `nuxt` or other `unjs`-based apps), is the application can be pre-rendered as static web assets (`html`, `css`, `js`, ...) and deployed to an AWS. To set it up one has to:
- configure the S3 bucket for web hosting (e.g. https://docs.aws.amazon.com/AmazonS3/latest/userguide/HostingWebsiteOnS3Setup.html) and transfer files to that bucket (typically via AWS CLI).

Nitro could deal with this case systematically, as proposed for Vercel (https://github.com/unjs/nitro/issues/818).

### Packaging Nitro for CloudFront's Lambda@Edge
Just like [Vercel's edge functions](https://vercel.com/docs/concepts/functions/edge-functions), Lambda@edge allows the developer to achieve better performance by exploiting a leaner runtime + a CDN-based caching.

-  Requires to set up a CloudFront with Lambda@edge environment

A proposal for an edge-based preset is available here (https://github.com/unjs/nitro/pull/1075/files) in response to the related issue https://github.com/unjs/nitro/issues/79.

### Amplify-based deployment
Amplify is a minimal-code frontend for configuring AWS services. It typically delivers one of the three previous options, although it scaffolds the services automatically based on a few input configurations from the developer.  

It does not currently provide a guide for Nuxt 3, although it does for [Nuxt 2](https://docs.amplify.aws/guides/hosting/nuxt/q/platform/js/#getting-started). 
I would leave this as a last-resort option for the moment, though, as it is still very rigid in terms of configurability (e.g. it does not allow to use Github action instead of AWS's built-in solution).
Related issue: https://github.com/unjs/nitro/issues/80

## CI/CD & docs
It would be nice to provide CI/CD examples as Github actions/ CodeBuild specs, as well as to extend the docs to guide the developers in the deployments

**Comment by ennioVisco** [comment]
@pi0 is this thread anyway useful for you and the development team?

**Reply by Hebilicious** [reply]
That's a great write-up, thanks @ennioVisco. We can use it to improve the docs once we get these 2 `aws-lambda-edge` presets merged.
